packages/cli/src/execution-lifecycle/__tests__/save-execution-progress.test.ts

```diff
@@ -1,100 +1,149 @@
 import { ErrorReporter } from 'n8n-core';
 import { Logger } from 'n8n-core';
-import type { IRunExecutionData, ITaskData, IWorkflowBase } from 'n8n-workflow';
+import type { IRunExecutionData, ITaskData } from 'n8n-workflow';
 
 import { ExecutionRepository } from '@/databases/repositories/execution.repository';
 import type { IExecutionResponse } from '@/interfaces';
 import { mockInstance } from '@test/mocking';
 
 import { saveExecutionProgress } from '../save-execution-progress';
-import * as fnModule from '../to-save-settings';
 
-mockInstance(Logger);
-const errorReporter = mockInstance(ErrorReporter);
-const executionRepository = mockInstance(ExecutionRepository);
+describe('saveExecutionProgress', () => {
+	mockInstance(Logger);
+	const errorReporter = mockInstance(ErrorReporter);
+	const executionRepository = mockInstance(ExecutionRepository);
 
-afterEach(() => {
-	jest.clearAllMocks();
-});
+	afterEach(() => {
+		jest.resetAllMocks();
+	});
+
+	const commonArgs: [string, string, string, ITaskData, IRunExecutionData] = [
+		'some-workflow-id',
+		'some-execution-id',
+		'My Node',
+		{} as ITaskData,
+		{} as IRunExecutionData,
+	];
 
-const commonArgs: [IWorkflowBase, string, string, ITaskData, IRunExecutionData, string] = [
-	{} as IWorkflowBase,
-	'some-execution-id',
-	'My Node',
-	{} as ITaskData,
-	{} as IRunExecutionData,
-	'some-session-id',
-];
-
-const commonSettings = { error: true, success: true, manual: true };
-
-test('should ignore if save settings say so', async () => {
-	jest.spyOn(fnModule, 'toSaveSettings').mockReturnValue({
-		...commonSettings,
-		progress: false,
+	test('should not try to update non-existent executions', async () => {
+		executionRepository.findSingleExecution.mockResolvedValue(undefined);
+
+		await saveExecutionProgress(...commonArgs);
+		expect(executionRepository.updateExistingExecution).not.toHaveBeenCalled();
 	});
 
-	await saveExecutionProgress(...commonArgs);
+	test('should handle DB errors on execution lookup', async () => {
+		const error = new Error('Something went wrong');
+		executionRepository.findSingleExecution.mockImplementation(() => {
+			throw error;
+		});
 
-	expect(executionRepository.updateExistingExecution).not.toHaveBeenCalled();
-});
+		await saveExecutionProgress(...commonArgs);
 
-test('should ignore on leftover async call', async () => {
-	jest.spyOn(fnModule, 'toSaveSettings').mockReturnValue({
-		...commonSettings,
-		progress: true,
+		expect(executionRepository.updateExistingExecution).not.toHaveBeenCalled();
+		expect(errorReporter.error).toHaveBeenCalledWith(error);
 	});
 
-	executionRepository.findSingleExecution.mockResolvedValue({
-		finished: true,
-	} as IExecutionResponse);
+	test('should handle DB errors when updating the execution', async () => {
+		const error = new Error('Something went wrong');
+		executionRepository.findSingleExecution.mockResolvedValue({} as IExecutionResponse);
```

Comment:
```suggestion
		executionRepository.findSingleExecution.mockResolvedValue(mock<IExecutionResponse>());
```

Reply:
couldn't get it to work because of all the deep proxies.